Automattic/jetpack
# Implemented pre-connection JITMs

This is a Jetpack plugin-only (pre-connection) port of the WPCOM JITM engine, aimed to present messages to users mostly to advertise Jetpack's features and encourage them to connect.

#### Changes proposed in this Pull Request:
* Implement a Jetpack-only (pre-connection) port of the WPCOM JITM engine to display JITMs without "phoning home".

#### Is this a new feature or does it add/remove features to an existing part of Jetpack?
* This is a new feature, mostly discussed in pbtFPC-nZ-p2 and related posts.

#### Testing instructions:
1) Get a brand new Jetpack environment, activate the Jetpack plugin but DO NOT connect it.
2) Currently there are 3 JITMs implemented **AS EXAMPLES** as specified in pbtFPC-nZ-p2, so **their text could change in the future before the release, or there could be more added**. They should be displayed at edit.php, upload.php, widgets.php like this:

![preconnection](https://user-images.githubusercontent.com/2166135/79801384-6d174100-8334-11ea-8737-4f3624a45310.png)

3) In addition to the JITMs rendering, click the blue "Setup Jetpack" buttons and check it takes you to the connection screen.
 
For a complete test:
1) Apply D42181-code.
2) Connect Jetpack.
3) You should now be able to see all the post-connection JITMs features already available and NOT the pre-connection ones.

#### Proposed changelog entry for your changes:
* Implemented pre-connection JITMs.
